src/main/java/com/emc/ecs/sync/storage/s3/AwsS3Storage.java

```diff
@@ -78,12 +81,39 @@
     public void configure(SyncStorage source, Iterator<SyncFilter> filters, SyncStorage target) {
         super.configure(source, filters, target);
 
-        Assert.hasText(config.getAccessKey(), "accessKey is required");
-        Assert.hasText(config.getSecretKey(), "secretKey is required");
+        if (!(config.getUseDefaultCredentialsProvider() || config.getProfile() != null)
+            || config.getSessionToken() != null
+            || config.getAccessKey() != null
+            || config.getSecretKey() != null) {
+            Assert.hasText(config.getAccessKey(), "accessKey is required");
+            Assert.hasText(config.getSecretKey(), "secretKey is required");
+            Assert.isTrue(config.getSessionToken() == null || config.getSessionToken().length() == 0,
+                          "sessionToken can not be empty");
+        }
+        Assert.isTrue(config.getProfile() == null || config.getProfile().length() > 0,
+            "Profile can not be empty");
         Assert.hasText(config.getBucketName(), "bucketName is required");
         Assert.isTrue(config.getBucketName().matches("[A-Za-z0-9._-]+"), config.getBucketName() + " is not a valid bucket name");
 
-        AWSCredentials creds = new BasicAWSCredentials(config.getAccessKey(), config.getSecretKey());
+        AwsS3CredentialsProviderChain.Builder providerChainBuilder = AwsS3CredentialsProviderChain.builder();
+
+        if (config.getSessionToken() != null && config.getSessionToken().length() > 0) {
+            providerChainBuilder.addCredentials(
+                new BasicSessionCredentials(config.getAccessKey(), config.getSecretKey(), config.getSessionToken()));
+        }
+        else if (config.getAccessKey() != null && config.getAccessKey().length() > 0) {
+            providerChainBuilder.addCredentials(
+                new BasicSessionCredentials(config.getAccessKey(), config.getSecretKey(), config.getSessionToken()));
```

Comment:
This is strange.  If `getSessionToken()` is null then this should be 

```
providerChainBuilder.addCredentials(
                new BasicAWSCredentials(config.getAccessKey(), config.getSecretKey());
```

Otherwise the old credentials method will be broken.  I don't know how I missed this before.